### PR TITLE
Adds missing functions

### PR DESCRIPTION
**Pull request by marrony**
this pr includes

* `abort` function
* `casefold:` with `normalizer` parameter
* deprecate `next_id` function
* `new_id` function
* `singleton` function
* `events` function
* `identity` function
* `has_identity` function

### CODE REVIEW

File: lib/fauna/query.rb [hunk 1]
```diff
@@ -460,6 +484,22 @@ def identify(ref, password)
       Expr.new identify: Expr.wrap(ref), password: Expr.wrap(password)
     end
 
+    ##
+    # An identity function
+    #
+    # Reference: {FaunaDB Authentication}[https://fauna.com/documentation/queries#auth_functions]
+    def identity()
```
Comment (ashfire908) [inline lib/fauna/query.rb]: You can omit the `()` if the method takes no arguments.

File: lib/fauna/query.rb [hunk 2]
```diff
@@ -460,6 +484,22 @@ def identify(ref, password)
...
+      Expr.new identity: nil
+    end
+
+    ##
+    # A has_identity function
+    #
+    # Reference: {FaunaDB Authentication}[https://fauna.com/documentation/queries#auth_functions]
+    def has_identity()
```
Comment (ashfire908) [inline lib/fauna/query.rb]: Ditto.

File: spec/query_spec.rb [hunk 3]
```diff
@@ -623,7 +666,6 @@ def test_method
         token = client.query { login @user[:ref], password: @password }
         user_client = get_client secret: token[:secret]
 
-        #fixme: use identity() instead
```
Comment (ashfire908) [inline spec/query_spec.rb]: How come this was removed and not performed?

File: CHANGELOG [hunk 4]
```diff
@@ -1,6 +1,12 @@
 3.0.0.pre.1
-* Adds support for recursive references
-* Removed REST api support from `Client`
+* Adds support for recursive references.
+* Removed REST api support from `Client`.
+* Added `abort` function.
+* Added `normalizer` argument to `casefold` function.
+* Deprecated `next_id` function.
```
Comment (ashfire908) [inline CHANGELOG]: Can we leave a note on this line that explicitly states `next_id` is replaced by `new_id`?

File: spec/query_spec.rb (hunk 3 above)
Comment (marrony) [reply spec/query_spec.rb]: when I added this comment I thought `identity` would return the token itself, but it's not the case, `identity` returns the instance associated with the token

**Pull request merged**